src/Rendering/utilityLayerRenderer.ts

```diff
@@ -200,9 +200,26 @@ export class UtilityLayerRenderer implements IDisposable {
                             scenePick = new PickingInfo();
                         }
                     } else {
+                        let previousActiveCamera: Nullable<Camera> = null
+                        // If a camera is set for rendering with this layer
+                        // it will also be used for the ray computation
+                        // To preserve back compat and because scene.pick always use activeCamera
+                        // it's substituted temporarily and a new scenePick is forced.
+                        // otherwise, the ray with previously active camera is always used.
+                        // It's set back to previous activeCamera after operation.
+                        if (this._renderCamera)
+                        {
+                            previousActiveCamera = scene.activeCamera;
+                            scene.activeCamera = this._renderCamera;

```

Comment:
You should update `activeCamera` using `_activeCamera` instead, else the `onActiveCameraChanged` notification will be fired.